joomla/joomla-cms
# Radio field: fix readonly and disabled option

Pull Request for Issue #8996 .
### Summary of Changes

After this PR radio fields can be `readonly` and single options of it can be `disabled`. Furthermore I adjusted the solution from @dgt41 in #10969 to fit with the solution of `disabled` options and `readonly`.
### Testing Instructions
- Take some radio fields and add the `disabled` or `readonly` attribute or add the `disabled` attribute to a single option.
- add the class `btn-group` and test again (In Beez3 you won't see a difference as the css code for `btn-group` is missing. See #12029.)

(I've applyed disabled, readonly and disabled option for illustrating. For testing please use only one at time since they have similar effects.)

``` xml
<field
    type="radio"
    default="1"
    disabled="true"
    readonly="true"
    class="btn-group"
>
    <option value="1">JYES</option>
    <option value="0" disabled="true">JNO</option>
</field>
```
### Documentation Changes Required

none

Btw.: Why isn't the JavaScript code related to `btn-group` in a central file so that all templates can use it without copying it? A central file would result in one file to update instead applying the same change on multiple files by multiple developers because every template developer must apply the fix himself.
